### PR TITLE
XiaoW_new timer refactor

### PR DESCRIPTION
**Pull request by wang9hu**
# Description
This is the refactored new timer frontend, and the related files were based on PR [#450](https://github.com/OneCommunityGlobal/HGNRest/pull/450#issue-1811283656). 
The new timer is based on Websocket connection communication and it is a count down timer instead of stop watch timer like the old timer.

## Related PRS (if any):
This frontend PR is related to the [#1282](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1282#issue-1886998437) frontend PR.
To test this frontend PR you need to checkout the [#1282](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1282#issue-1886998437) frontend PR.

## How to test:
1. check into current branch
2. check into frontend branch
3. do `npm install` and `npm run dev` to run this PR locally
4. follow the instructions on the frontend branch

## Note:
1. `npm run dev` will automatically build and start the server, and rebuild and restart when any changes is saved in `/src` directory, so there is no need to run `npm run build` along with `npm start` any more.

### CODE REVIEW

**Review by navneeeth — APPROVED** [review]
Hello! Tested your changes and they LGTM! I have commented a detailed review in the frontend PR.

**Review by StrawberryCalpico — APPROVED** [review]
I tested your PR, it works great! Details in [FE#1282](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1282#pullrequestreview-1686197315)

**Review by malikjahanzaib — APPROVED** [review]
Hi @wang9hu, I have reviewed this PR and everything seems to work fine. Left detailed review on FE PR. Nice work!!!

**Review by cvtqx — DISMISSED** [review]
Requested changes in the front end PR.

**Review by Shubhu0 — APPROVED** [review]
Tested and reviewed your PR. Details on the FE PR #1282

**Review by eduhorta — COMMENTED** [review]
Left a more detailed review on the FE PR.
Great job! 👍

**Review by shubhankarval — DISMISSED** [review]
Hey @wang9hu, I am requesting changes, since I got the following errors on the backend and frontend while running the app. 
Backend:
pr
Console:
pr1

**Review by savio-henrique — APPROVED** [review]
Hi @wang9hu , this PR was tested and reviewed! More details on [FE#1282](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1282) !

**Review by ChirayuRai — COMMENTED** [review]
Added my comments on FE#1282

**Comment by RheaWu1212** [comment]
Tested and reviewed your PR. Details on the FE PR #1282

**Review by yaow62023 — APPROVED** [review]
This PR are test at frontend branch #1282.

**Comment by palakgosalia** [comment]
Hey, Reviewed this PR and tested it. Details in #1282

File: src/websockets/index.js [hunk 1]
```diff
@@ -118,64 +73,66 @@ export default async (expServer) => {
...
+      }
       ws.isAlive = false;
       ws.ping();
     });
-  }, 3000000);
+  }, 10000);
 
-  // Here we just clear the interval when the server closes
-  wss.on("close", () => {
+  wss.on('close', () => {
```
Comment (John-Flavian) [inline src/websockets/index.js:135]: Nice work.

However, can it be modified to automatically save the last state of the user's timer when the connection has been terminated; such that when the user logs in again; he/she resumes from their last state.

Comment (wang9hu) [reply src/websockets/index.js:135]: Thank you for your advice! The feature you mentioned is implemented in the clientHandler file, in which every changes would be saved to the database.

Comment (John-Flavian) [reply src/websockets/index.js:135]: Awesome.

**Pull request merged**